src/rules/flake8_no_pep420/rules.rs

```diff
@@ -20,7 +20,7 @@ impl Violation for ImplicitNamespacePackage {
 
 /// INP001
 pub fn implicit_namespace_package(path: &Path, package: Option<&Path>) -> Option<Diagnostic> {
-    if package.is_none() {
+    if package.is_none() && path.extension().map_or(false, |ext| ext == "py") {
```

Comment:
```suggestion
    if package.is_none() && ! path.extension().map_or(false, |ext| ext == "pyi") {
```
otherwise I guess all files without file extensions are ignored?

Reply:
Hmm -- fair!